src/tagmanager/tm_ctags.c

```diff
@@ -241,6 +241,33 @@ void tm_ctags_clear_ignore_symbols(void)
 }
 
 
+static gboolean replace_str(gchar **where, const gchar *what, gsize what_len,
+	const gchar *replacement, gsize replacement_len)
+{
+	if (where && *where)
+	{
+		gchar *pos = strstr(*where, what);
+
+		if (pos)
+		{
+			gsize where_len = strlen(*where);
+			gchar *str = g_malloc(where_len + (replacement_len - what_len) + 1);
```

Comment:
hum, the issue might be the parentheses here: as those are both unsigned, wouldn't it wrap around if `what_len < replacement_len`? if that's it, just dropping the parentheses should work (as `what_len` cannot be longer than `where_len`, and `replacement_len` cannot be negative)

Reply:
sorry to have created a mess in your PR…

Reply:
Ah, I only noticed your message now. The gsize change I made should be alright too, right?

Reply:
I don't see why they would? Admittedly the tests passed, but I can't see why using a larger unsigned would solve the underflow… unless maybe it subsequently overflows in final computation, putting sign back in the right direction?
even if that's so, it's a bit too scary even for me :)

Reply:
…yeah that's what happens, so I'd rather just drop the parentheses if you're fine with it :)

Reply:
Ah, OK,  gsize isn't signed...

Anyway, I just tried to put guints back in the function parameters, dropped the parentheses, and... the `nested_anon.c` test failed. Something somewhere still isn't right.

Reply:
> Anyway, I just tried to put guints back in the function parameters, dropped the parentheses, and... the nested_anon.c test failed. Something somewhere still isn't right.

Nah, forget it, I modified the test locally, it works alright.

Reply:
ah, good!  I tested it locally and couldn't get it to fail without the parentheses, so I was starting to loose a lot of hairs over that :laughing:

Reply:
Yeah, sorry for your hair loss :-)